helpers/controllers/submissionController.ts

```diff
@@ -9,18 +9,38 @@ import {
   SubmissionStatus
 } from '../../graphql'
 import prisma from '../../prisma'
-import { decode } from '../encoding'
 import { hasPassedLesson } from '../hasPassedLesson'
 import { updateSubmission } from '../updateSubmission'
 import { sendSubmissionNotification, IdType } from '../discordBot'
+import { decode } from '../encoding'
+import * as Sentry from '@sentry/node'
+import { CliToken } from '../../@types/user'
 
 export const createSubmission = async (
   _parent: void,
-  args: MutationCreateSubmissionArgs
+  args: MutationCreateSubmissionArgs,
+  ctx: Context
 ): Promise<CreateSubmissionMutation['createSubmission']> => {
+  const { req } = ctx
+
   if (!args) throw new Error('Invalid args')
   const { challengeId, cliToken, diff, lessonId } = args
-  const { id } = decode(cliToken)
+
+  const decodedCliToken: CliToken = cliToken && decode(cliToken)
+  let id = !req.user ? decodedCliToken.id : req.user.id
```

Comment:
bad coding practice to have double negatives: `!req.user` unless you absolutely have to, which is not the case here.

```js
let id = _.get(req, 'user.id')
if (!id && cliToken) {
  const user = ...
  id = user!?.id
}
```